hack/make-rules/update.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A single script that runs a predefined set of update-* scripts, as they often go together.
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# If called directly, exit.
+if [[ "${CALLED_FROM_MAIN_MAKEFILE:-""}" == "" ]]; then
+    echo "ERROR: $0 should not be run directly." >&2
+    echo >&2
+    echo "Please run this command using \"make update\""
+    exit 1
+fi
+
+SILENT=${SILENT:-true}
+ALL=${FORCE_ALL:-false}
+V=""
+if [[ "${SILENT}" != "true" ]]; then
+	V="-v"
+fi
+
+trap 'exit 1' SIGINT
+
+if $SILENT ; then
+	echo "Running in silent mode, run with SILENT=false if you want to see script logs."
+fi
+
+if ! $ALL ; then
+	echo "Running in short-circuit mode; run with FORCE_ALL=true to force all scripts to run."
+fi
+
+"${KUBE_ROOT}/hack/godep-restore.sh" ${V}
+
+BASH_TARGETS="
+	update-generated-protobuf
+	update-codegen
+	update-generated-runtime
+	update-generated-device-plugin
+	update-generated-docs
+	update-generated-swagger-docs
+	update-swagger-spec
+	update-openapi-spec
+	update-api-reference-docs
+	update-staging-godeps
+	update-bazel"
+
+for t in $BASH_TARGETS; do
+	echo -e "${color_yellow}Running $t${color_norm}"
+	if $SILENT ; then
+		if ! bash "$KUBE_ROOT/hack/$t.sh" 1> /dev/null; then
```

Comment:
nit: `${KUBE_ROOT}`

Reply:
Can we pipe the stderr output into a file and show it in case of error?

Reply:
Also RE: stderr, we aren't piping it anywhere, so it will be displayed already:
```
$ cat hack/update-test.sh 
echo "Test error message" 1>&2
exit 1
$ make update 
Running in silent mode, run with SILENT=false if you want to see script logs.
Running in short-circuit mode; run with FORCE_ALL=true to force all scripts to run.
+++ [0122 15:07:18] Restoring kubernetes godeps
+++ [0122 15:07:22] Dependencies appear to be current - skipping download
Running update-test
Test error message
Running update-test FAILED
make: *** [update] Error 1
```